samza-core/src/test/scala/org/apache/samza/util/TestExponentialSleepStrategy.scala

```diff
@@ -116,45 +118,57 @@ class TestExponentialSleepStrategy {
     assertEquals(0, loopObject.sleepCount)
   }
 
-  def interruptedThread(operation: => Unit) = {
+  def interruptedThread(latch: CountDownLatch, operation: => Unit): Option[Throwable] = {
     var exception: Option[Throwable] = None
     val interruptee = new Thread(new Runnable {
       def run {
         try { operation } catch { case e: Exception => exception = Some(e) }
       }
     })
-    interruptee.start
-    Thread.sleep(10) // give the thread a chance to make some progress before we interrupt it
-    interruptee.interrupt
-    interruptee.join
+    interruptee.start()
+    latch.await()
```

Comment:
Would prefer the following so the test doesn't hang

assertTrue("Latch timed out", latch.await(someHighNumber, someUnit))